nginx-jwt.lua

```diff
@@ -0,0 +1,44 @@
+local jwt = require "resty.jwt"
+local cjson = require "cjson"
+local secret = os.getenv("JWT_SECRET")
+
+assert(secret ~= nil, "Environment variable JWT_SECRET not set")
+
+local M = {}
+
+function M.auth()
+    -- require Authorization request header
+    local auth_header = ngx.var.http_Authorization
+
+    if auth_header == nil then
+        ngx.log(ngx.WARN, "No Authorization header")
+        ngx.exit(ngx.HTTP_UNAUTHORIZED)
+    else
+        ngx.log(ngx.INFO, "Authorization: " .. auth_header)
+
+        -- require Bearer token
+        local _, _, token = string.find(auth_header, "Bearer%s+(.+)")
+
+        if token == nil then
+            ngx.log(ngx.WARN, "Missing token")
+            ngx.exit(ngx.HTTP_UNAUTHORIZED)
+        else
+            ngx.log(ngx.INFO, "Token: " .. token)
+
+            -- require valid JWT
+            local jwt_obj = jwt:verify(secret, token)
+            if jwt_obj.verified == false then
+                ngx.log(ngx.WARN, "Invalid token: ".. jwt_obj.reason)
+                ngx.exit(ngx.HTTP_UNAUTHORIZED)
+            else
+                ngx.log(ngx.INFO, "JWT: " .. cjson.encode(jwt_obj))
+
+                -- write the X-Auth-UserId header 
+                ngx.header["X-Auth-UserId"] = jwt_obj.payload.sub
```

Comment:
Agreed.  And the only reason I even send back the X-Auth-UserId header in the first place is so that the [integration test](https://github.com/twistedstream/nginx-jwt/blob/issue-4/test/test_integration.js#L37) can assert that the JWT was actually verified.